packages/lib-runtime/src/store/PluginStore.ts

```diff
@@ -0,0 +1,309 @@
+import * as _ from 'lodash-es';
+import { Extension, LoadedExtension } from '../types/extension';
+import { PluginMetadata, PluginManifest, LoadedPlugin } from '../types/plugin';
+import { consoleLogger } from '../utils/logger';
+import { PluginLoader } from './PluginLoader';
+
+/**
+ * Client interface for `PluginStore` consumers.
+ */
+export interface PluginStoreClient<TLoadedExtension extends LoadedExtension> {
+  subscribe: (listener: VoidFunction, eventTypes: PluginStoreEventType[]) => VoidFunction;
+  getExtensions: () => TLoadedExtension[];
+  getPluginInfo: () => (LoadedPluginInfo | NotLoadedPluginInfo)[];
```

Comment:
I think any type exposed on a public type should themselves be exposed.

Reply:
Might be good to establish some guidelines when to export types for consistency across the SDK. That could be a longer discussion and shouldn't hold up this PR in the interim.

Reply:
cc @andrewballantyne  ^^